### DISCUSSION TITLE
MAUI should learn from Avalonia

### DISCUSSION THREAD

**Discussion by akhanalcs** [dotnet/maui#7779] [Ideas]
**Edit:** I changed my mind about drawn controls and support native controls offered by MAUI. So rest of this post doesn't really resonate with me anymore.

---

For MAUI to stay competitive in cross platform app development, it'd definitely help to learn from [Avalonia](https://avaloniaui.net/), and forgo native controls in favor of renderers and also learn from their [clean and simple architecture](https://dev.to/avalonia/avalonia-platform-support-why-its-simple-cjd). 

Avalonia has solved the cross platform problem so elegantly. Why not learn from them?

**Comment by TonyK-FmGroup** [comment]
Having never heard of Avalonia, I naturally followed the link. if anybody can work out what this thing is from just the home page, I would be amazed. Three of us eventually worked it out from the documentation and quickly returned to MAUI.

**Reply by Symbai** [reply]
> if anybody can work out what this thing is

In short: WPF but cross platform. It is very easy to create small applications for Windows, Linux and MacOS and kinda stable enough for that purpose. But it only has a small team and some issues might not get addressed for years. Cannot say anything about mobile development as I have no experience with Avalonia on that.

**Reply by TonyK-FmGroup** [reply]
It seemed to be more a XAML based framework but not actually WPF so would only be limited to the actual controls provided by the framework. But, that's just what we think we worked out from the documentation. Whereas Platform.Uno actually renders WPF/UWP to the multiple platforms which means we are able to use useful things like the community toolkit which looks like it wouldn't be possible with Avalonia.

**Reply by TonyK-FmGroup** [reply]
Actually, taking a closer look, this looks very interesting indeed, we're wondering why we've never heard of this before now. It does now somewhat complicate our decision into which UI tech to use for a greenfield project with a new player in the park!

**Comment by montyclt** [comment]
You're free to use Avalonia instead of MAUI if for your project native controls are not important. I love to have the both projects active, and select MAUI or Avalonia depending if I need native controls.

For me, I native controls are mandatory in some projects: https://github.com/dotnet/maui/discussions/1058#discussioncomment-1291130

**Comment by emilsteen** [comment]
> "and some issues might not get addressed for years."

Well, that is my experience with Xamarin. And that is why I'm looking for alternatives to MAUI.

**Comment by yurkinh** [comment]
Actually you can combine both and reuse Avalonia controls in  .net maui if you want;

https://github.com/AvaloniaUI/AvaloniaMauiHybrid

**Comment by pierre01** [comment]
I wonder sometimes if the effort MAUI put on using native controls would have cost much less time and effort by just maintaining custom drawn libraries of controls adapted for each Operating system like Avalonia.

**Comment by yurkinh** [comment]
Another very interesting and promising solution for .net maui with drawn controls is DrawnUi:
https://github.com/taublast/DrawnUi.Maui
